Changed some function to use multiple databases

hinashi opened the pull request:
Changed below to use multiple databases.

- Added the django-replicated library
- Removed my own DB selection logic
- Changed the logout from GET to POST method
- Removed complement_attrs when request show entry page
- Added API to create Attribute for edit ACL
- Added feature to return default Attribute value

userlocalhost commented on entry/models.py:
```diff
@@ -1116,26 +1092,42 @@ def complement_attrs(self, user):
+
                 if not last_value.data_type:
                     last_value.data_type = attr.schema.type
                     last_value.save()
```
DB への書き込みが発生しそうな予感...
仕様変更（get_available_attrs で DB への更新を行わない）に伴い、DB へのアクセスを伴う処理は無くした方が良いかと思います。
（当該ロジックは通らない想定なので、単純に消すだけでも問題ないかと思います）

takano32 commented on airone/auth/view.py:
```diff
@@ -0,0 +1,14 @@
+from airone.lib.http import render
+
+
+@csrf_protect
```
👍

userlocalhost commented:
Would you please correct this processing? Because it might be run from GET request event handler (history method, which is declared at entry/views.py)
https://github.com/dmm-com/airone/blob/9c535afc2b946ea7f95b33ac841ad9ba520498fa/entry/models.py#L178-L181

userlocalhost commented:
Those two methods in `Entry.complement_attrs()` might send queries to DB for update. And this method is called by `export()` method, which expects not to send update queries to DB. Could you please caring them?

https://github.com/dmm-com/airone/blob/9c535afc2b946ea7f95b33ac841ad9ba520498fa/entry/models.py#L1074

https://github.com/dmm-com/airone/blob/9c535afc2b946ea7f95b33ac841ad9ba520498fa/entry/models.py#L1093

hinashi commented:
@userlocalhost 
Thank you for checking.
I forgot that export was changed from GET to POST in the PR below.
https://github.com/dmm-com/airone/pull/170

Changed entry model export from database read to write.

hinashi commented:
@userlocalhost 
I removed the DB write logic from get_value_history.
The data_type is always set, so it looks like unnecessary logic.

In addition, I deleted the unused DB selection logic because it remained.

userlocalhost commented:
Thank you very much for your hard work!
